gc.md

```diff
@@ -1,65 +1,78 @@
-      - [Sweep and Compact](#sweep-and-compact)
-      - [Sweep](#sweep)
-      - [Compact](#compact)
-    - [Incremental Mark and Lazy Sweep](#incremental-mark-and-lazy-sweep)
-      - [Incremental Marking](#incremental-marking)
-      - [Lazy Sweeping](#lazy-sweeping)
-  - [Causes For GC Pause](#causes-for-gc-pause)
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Goals, Techniques](#goals-techniques)
+- [Cost of Allocating Memory](#cost-of-allocating-memory)
+- [How objects are determined to be dead](#how-objects-are-determined-to-be-dead)
+- [Two Generations](#two-generations)
+  - [Generational Garbage Collector](#generational-garbage-collector)
+- [Heap Organization in Detail](#heap-organization-in-detail)
+  - [New Space aka Young Generation](#new-space-aka-young-generation)
+  - [Old Pointer Space](#old-pointer-space)
+  - [Old Data Space](#old-data-space)
+  - [Large Object Space](#large-object-space)
+  - [Code Space](#code-space)
+  - [Cell Space, Property Cell Space, Map Space](#cell-space-property-cell-space-map-space)
+  - [Pages](#pages)
+- [Young Generation](#young-generation)
+  - [ToSpace, FromSpace, Memory Exhaustion](#tospace-fromspace-memory-exhaustion)
+    - [Sample Scavenge Scenario](#sample-scavenge-scenario)
+      - [Collection to free ToSpace](#collection-to-free-tospace)
+    - [Considerations](#considerations)
+- [Orinoco Garbage Collector](#orinoco-garbage-collector)
+  - [Parallel Scavenger](#parallel-scavenger)
+    - [Scavenger Phases](#scavenger-phases)
+      - [Distribution of scavenger work across one main thread and two worker threads](#distribution-of-scavenger-work-across-one-main-thread-and-two-worker-threads)
+    - [Results](#results)
+  - [Techniques to Improve GC Performance](#techniques-to-improve-gc-performance)
+    - [Memory Partition and Parallelization](#memory-partition-and-parallelization)
+    - [Tracking Pointers](#tracking-pointers)
+    - [Black Allocation](#black-allocation)
   - [Resources](#resources)
+- [Old Generation Garbage Collector Deep Dive](#old-generation-garbage-collector-deep-dive)
+  - [Collection Steps](#collection-steps)
+  - [Mark Sweep and Mark Compact](#mark-sweep-and-mark-compact)
+    - [Mark](#mark)
+    - [Marking State](#marking-state)
+      - [Depth-First Search](#depth-first-search)
+    - [Handling Deque Overflow](#handling-deque-overflow)
+    - [Sweep and Compact](#sweep-and-compact)
+    - [Sweep](#sweep)
+    - [Compact](#compact)
+  - [Incremental Mark and Lazy Sweep](#incremental-mark-and-lazy-sweep)
+    - [Incremental Marking](#incremental-marking)
+    - [Lazy Sweeping](#lazy-sweeping)
+- [Resources](#resources-1)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-# v8 Garbage Collector
-
 ## Goals, Techniques
 
 - ensures fast object allocation, short garbage collection pauses and no memory fragmentation
 - **stop-the-world**,
```

Comment:
Is this about Mark-Compact GC or Scavenger?

Mark-Compact GC is not stop-the-world. It was incremental since 2011. Recently it became mostly concurrent (blog post is coming).

Reply:
This came from the description on the [V8 wiki](https://github.com/v8/v8/wiki/Design%20Elements#efficient-garbage-collection).

But I'd assume this is more important regarding scavenging since that happens much more often and usually is the reason for perf bottlenecks due to GC.
From what I understand it's mostly parallel now as well, but there are still steps for which it has to _stop the world_ to ensure nothing is allocated on the heap while it's performing those steps.

Is that correct and sufficient to call it _stop the world_?
If so I'll just clarify that this mostly applies to the scavenger.

Reply:
Tried to address in 9d79c90.
I tried to make it more obvious in the Goals section which parts actually stop-the-world by adding a few more notes and linking to parts of the document where I explain incremental mark-compact GC.

Reply:
Yep, the scanveger is a stop-the-world parallel GC.

Thanks for the pointer. The V8 wiki needs update.